keras_hub/src/models/cspnet/cspnet_backbone.py

```diff
@@ -0,0 +1,1269 @@
+            - the first two convolutions of `"bottleneck_block"`
+            of each stage. Defaults to `1.0`.
+        block_ratio: float or tuple/list of floats. Filter size for each block,
+            calculated as `(stackwise_num_filters * block_ratio)` for each
+            stage. Defaults to `1.0`.
+        expand_ratio: float or tuple/list of floats. Filters ratio for `"csp"`
+            and `"cs3"` stages at different levels. Defaults to `1.0`.
+        stem_padding: str, padding value for the stem, either `"valid"` or
+            `"same"`. Defaults to `"valid"`.
+        stem_pooling: str, pooling value for the stem. Defaults to `None`.
+        avg_down: bool, if `True`, `AveragePooling2D` is applied at the
+            beginning of each stage when `strides == 2`. Defaults to `False`.
+        down_growth: bool, grow downsample channels to output channels. Applies
+            to Cross Stage only. Defaults to `False`.
+        cross_linear: bool, if `True`, activation will not be applied after the
+            expansion convolution. Applies to Cross Stage only. Defaults to
+            `False`.
+        data_format: `None` or str. If specified, either `"channels_last"` or
+            `"channels_first"`. The ordering of the dimensions in the inputs.
+            `"channels_last"` corresponds to inputs with shape
+            `(batch_size, height, width, channels)` while `"channels_first"`
+            corresponds to inputs with shape
+            `(batch_size, channels, height, width)`. It defaults to the
+            `image_data_format` value found in your Keras config file at
+            `~/.keras/keras.json`. If you never set it, then it will be
+            `"channels_last"`.
+        image_shape: tuple. The input shape without the batch size.
+            Defaults to `(None, None, 3)`.
+        dtype: `None` or str or `keras.mixed_precision.DTypePolicy`. The dtype
+            to use for the model's computations and weights.
+
+    Examples:
+    ```python
+    input_data = np.ones(shape=(8, 224, 224, 3))
+
+    # Pretrained backbone
+    model = keras_hub.models.CSPNetBackbone.from_preset(
+        "cspdarknet53_ra_imagenet"
+    )
+    model(input_data)
+
+    # Randomly initialized backbone with a custom config
+    model = keras_hub.models.CSPNetBackbone(
+        stem_filters=32,
+        stem_kernel_size=3,
+        stem_strides=1,
+        stackwise_depth=[1, 2, 4],
+        stackwise_strides=[1, 2, 2],
+        stackwise_num_filters=[32, 64, 128],
+        block_type="dark,
+    )
+    model(input_data)
+    ```
+    """
+
+    def __init__(
+        self,
+        stem_filters,
+        stem_kernel_size,
+        stem_strides,
+        stackwise_depth,
+        stackwise_strides,
+        stackwise_num_filters,
+        block_type,
+        groups=1,
+        stage_type=None,
+        activation="leaky_relu",
+        output_strides=32,
+        bottle_ratio=[1.0],
+        block_ratio=[1.0],
```

Comment:
are any of these default values similar across different presets?
if it is - it would be great to reduce the number of args here and set their value in the code directly

Reply:
for many of the presets, activation will be `leaky_relu`, if users wants to add custom activation like the standard `relu`, we kept it.